### PR TITLE
32 migrate to hal 10

### LINKED ISSUE #32: Migrate to HAL 1.0
It's time to migrate to embedded-hal v1

### PR DESCRIPTION
**Pull request by HusseinAbdelhamid**

### CODE REVIEW

File: src/example.rs [hunk 1]
```diff
@@ -5,79 +5,86 @@ use alloc::vec;
...
+                            *val += (i + 1) as u8;
+                        });
+                        return Ok(());
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
```
Comment (marius-meissner) [inline src/example.rs:57]: That's quite a deep if nesting. It's example code, so it's already whining on a high level :)
Maybe you can make the statement a bit flatter with “ealry returns”.

**Review by marius-meissner — CHANGES_REQUESTED** [review]
Looks good :+1: 
Just one tiny suggestions (s. inline comment). 

Maybe you can in addition the following targets to QA pipeline:
* thumbv7m-none-eabi (Raspberry Pi)
* thumbv8m.main-none-eabihf (RP2350)
* riscv32imac-unknown-none-elf (RP2350 alternative)

**Comment by HusseinAbdelhamid** [comment]
Changes addressed now :+1:

**Review by marius-meissner — APPROVED** [review]
LGTM

**Pull request merged**